src/cascadia/TerminalApp/Command.cpp

```diff
@@ -11,14 +11,32 @@
 
 using namespace winrt::Microsoft::Terminal::Settings;
 using namespace winrt::TerminalApp;
+using namespace winrt::Windows::Foundation;
 
 static constexpr std::string_view NameKey{ "name" };
 static constexpr std::string_view IconPathKey{ "iconPath" };
 static constexpr std::string_view ActionKey{ "command" };
 static constexpr std::string_view ArgsKey{ "args" };
+static constexpr std::string_view IterateOnKey{ "iterateOn" };
+static constexpr std::string_view CommandsKey{ "commands" };
+
+static constexpr std::string_view IterateOnProfilesValue{ "profiles" };
+
+static constexpr std::wstring_view ProfileName{ L"${profile.name}" };

```

Comment:
whatever we choose, I absolutely want to make sure we can eventually give this a parser and a key/value reader so we don't need to specialize variable substitutions by the hundreds :smile:

Reply:
Yea, I could see that being a reasonable follow-up to this. I'd rather not convolute this PR more though, if possible 😉

Reply:
I mostly just mean “let’s choose a syntax that isn’t going to hamper our efforts here in the future” moreso than “do it in this PR” :)

Reply:
Oh okay, that makes sense.

We could maybe even drop the `profile.` here. You're already iterating on `profiles`. Maybe that's redundant?

Reply:
(I don’t want us to regret it later and have to shim toooooo much)

Reply:
I like that idea actually, profile being implied. Or like Go templates, `.name` where . Implies the object currently in scope (it supports referencing things that aren’t in scope)